File: src/ui/pathways.py
"""Age-based clinical pathway constants (<15 vs 15+)."""

from __future__ import annotations

AGE_BANDS: dict[str, int] = {
    "Under 2 months": 1,
    "2 months \u2013 5 years": 24,
    "5\u201315 years": 96,
    "15\u201317 years": 192,
    "18\u201364 years": 480,
    "65+ years": 840,
}

PATHWAY_CHILD = "Child (under 15)"
PATHWAY_ADULT = "Adult (15+)"

CHILD_AGE_BANDS_ORDERED: tuple[str, ...] = (
    "Under 2 months",
    "2 months \u2013 5 years",
    "5\u201315 years",
)
ADULT_AGE_BANDS_ORDERED: tuple[str, ...] = (
    "15\u201317 years",
    "18\u201364 years",
    "65+ years",
)

PEDIATRIC_AGE_BANDS: frozenset[str] = frozenset(CHILD_AGE_BANDS_ORDERED)
ADULT_AGE_BANDS: frozenset[str] = frozenset(ADULT_AGE_BANDS_ORDERED)

# Backward-compatible alias for comorbidity gating.
COMORBIDITY_AGE_BANDS: frozenset[str] = ADULT_AGE_BANDS


def age_bands_for_pathway(pathway_label: str) -> tuple[str, ...]:
    if pathway_label == PATHWAY_ADULT:
        return ADULT_AGE_BANDS_ORDERED
    return CHILD_AGE_BANDS_ORDERED


def default_age_band_index(pathway_label: str) -> int:
    """Default to under-5 for children, 18–64 for adults."""
    return 1


def is_pediatric_pathway(age_band: str) -> bool:
    return age_band in PEDIATRIC_AGE_BANDS


def is_adult_pathway(age_band: str) -> bool:
    return age_band in ADULT_AGE_BANDS

File: src/lib/fevergate/pathways.ts
import { mm } from "@/lib/i18n/mm";

export const AGE_BANDS: Record<string, number> = {
  [mm.age.under2Months]: 1,
  [mm.age.months2to5]: 24,
  [mm.age.years5to15]: 96,
  [mm.age.years15to17]: 192,
  [mm.age.years18to64]: 480,
  [mm.age.years65plus]: 840,
};

export const PATHWAY_CHILD = mm.age.pathwayChild;
export const PATHWAY_ADULT = mm.age.pathwayAdult;

export const CHILD_AGE_BANDS = [
  mm.age.under2Months,
  mm.age.months2to5,
  mm.age.years5to15,
] as const;

export const ADULT_AGE_BANDS = [
  mm.age.years15to17,
  mm.age.years18to64,
  mm.age.years65plus,
] as const;

export function ageBandsForPathway(pathway: string): readonly string[] {
  return pathway === PATHWAY_ADULT ? ADULT_AGE_BANDS : CHILD_AGE_BANDS;
}

export function defaultAgeBandIndex(_pathway: string): number {
  return 1;
}

export function isPediatricPathway(ageBand: string): boolean {
  return (CHILD_AGE_BANDS as readonly string[]).includes(ageBand);
}

export function isAdultPathway(ageBand: string): boolean {
  return (ADULT_AGE_BANDS as readonly string[]).includes(ageBand);
}
